File: GXpert/GXpert.Web/Modules/LiveSessions/Poll/PollForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.LiveSessions.Forms;

[FormScript("LiveSessions.Poll")]
[BasedOnRow(typeof(PollRow), CheckNames = true)]
public class PollForm
{

    public string Question { get; set; }
    [HalfWidth]
    public short EQuestionType { get; set; }
    [HalfWidth]
    public string Option1 { get; set; }
    [HalfWidth]
    public string Option2 { get; set; }
    [HalfWidth]
    public string Option3 { get; set; }
    [HalfWidth]
    public string Option4 { get; set; }
    [HalfWidth]
    public string Option5 { get; set; }
    [HalfWidth]
    public string RightAnswer { get; set; }
    [HalfWidth]
    public int LiveSessionLogId { get; set; }
    [HalfWidth]
    public short TotalAttempts { get; set; }
    [HalfWidth]
    public float AverageResponseTime { get; set; }
    [HalfWidth]
    public short NumberOfCorrect { get; set; }
    [HalfWidth]
    public short NumberOfWrong { get; set; }
    [HalfWidth]
    public int BloomsTaxonomyId { get; set; }
    [HalfWidth]
    public float Weightage { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/LiveSessions/PollForm.ts
import { StringEditor, EnumEditor, ServiceLookupEditor, IntegerEditor, DecimalEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EQuestionType } from "../Web/Enums.EQuestionType";

export interface PollForm {
    Question: StringEditor;
    EQuestionType: EnumEditor;
    Option1: StringEditor;
    Option2: StringEditor;
    Option3: StringEditor;
    Option4: StringEditor;
    Option5: StringEditor;
    RightAnswer: StringEditor;
    LiveSessionLogId: ServiceLookupEditor;
    TotalAttempts: IntegerEditor;
    AverageResponseTime: DecimalEditor;
    NumberOfCorrect: IntegerEditor;
    NumberOfWrong: IntegerEditor;
    BloomsTaxonomyId: LookupEditor;
    Weightage: DecimalEditor;
}

export class PollForm extends PrefixedContext {
    static readonly formKey = 'LiveSessions.Poll';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PollForm.init)  {
            PollForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = ServiceLookupEditor;
            var w3 = IntegerEditor;
            var w4 = DecimalEditor;
            var w5 = LookupEditor;

            initFormType(PollForm, [
                'Question', w0,
                'EQuestionType', w1,
                'Option1', w0,
                'Option2', w0,
                'Option3', w0,
                'Option4', w0,
                'Option5', w0,
                'RightAnswer', w0,
                'LiveSessionLogId', w2,
                'TotalAttempts', w3,
                'AverageResponseTime', w4,
                'NumberOfCorrect', w3,
                'NumberOfWrong', w3,
                'BloomsTaxonomyId', w5,
                'Weightage', w4
            ]);
        }
    }
}

[EQuestionType]; // referenced types